controllers/update_status_test.go

```diff
@@ -367,69 +358,55 @@ var _ = Describe("update_status", func() {
-				processGroup := cluster.Status.ProcessGroups[len(cluster.Status.ProcessGroups)-4]
-				Expect(processGroup.ProcessGroupID).To(Equal(storageOneProcessGroupID))
-				Expect(processGroup.ProcessGroupConditions).To(HaveLen(1))
-				Expect(processGroup.ProcessGroupConditions[0].ProcessGroupConditionType).To(Equal(fdbv1beta2.ProcessIsMarkedAsExcluded))
+
+				incorrectProcesses := fdbv1beta2.FilterByCondition(cluster.Status.ProcessGroups, fdbv1beta2.ProcessIsMarkedAsExcluded, false)
+				Expect(incorrectProcesses).To(ConsistOf([]fdbv1beta2.ProcessGroupID{pickedProcessGroup.ProcessGroupID}))
+				Expect(cluster.Status.ProcessGroups).To(HaveLen(17))
 			})
 		})
 
 		When("a process group has the wrong command line", func() {
 			BeforeEach(func() {
-				adminClient.MockIncorrectCommandLine(storageOneProcessGroupID, true)
+				adminClient.MockIncorrectCommandLine(pickedProcessGroup.ProcessGroupID, true)
 			})
 
 			It("should get a condition assigned", func() {
 				err := validateProcessGroups(context.TODO(), clusterReconciler, cluster, &cluster.Status, processMap, configMap, allPvcs, logger, "")
 				Expect(err).NotTo(HaveOccurred())
 
 				incorrectProcesses := fdbv1beta2.FilterByCondition(cluster.Status.ProcessGroups, fdbv1beta2.IncorrectCommandLine, false)
-				Expect(incorrectProcesses).To(Equal([]fdbv1beta2.ProcessGroupID{storageOneProcessGroupID}))
-
-				Expect(len(cluster.Status.ProcessGroups)).To(BeNumerically(">", 4))
-				processGroup := cluster.Status.ProcessGroups[len(cluster.Status.ProcessGroups)-4]
-				Expect(processGroup.ProcessGroupID).To(Equal(storageOneProcessGroupID))
-				Expect(len(processGroup.ProcessGroupConditions)).To(Equal(1))
+				Expect(incorrectProcesses).To(ConsistOf([]fdbv1beta2.ProcessGroupID{pickedProcessGroup.ProcessGroupID}))
+				Expect(cluster.Status.ProcessGroups).To(HaveLen(17))
 			})
 
 			When("the process group is marked for removal", func() {
 				BeforeEach(func() {
-					cluster.Spec.ProcessGroupsToRemove = []fdbv1beta2.ProcessGroupID{storageOneProcessGroupID}
+					cluster.Spec.ProcessGroupsToRemove = []fdbv1beta2.ProcessGroupID{pickedProcessGroup.ProcessGroupID}
 				})
 
 				It("should get a condition assigned", func() {
 					err := validateProcessGroups(context.TODO(), clusterReconciler, cluster, &cluster.Status, processMap, configMap, allPvcs, logger, "")
 					Expect(err).NotTo(HaveOccurred())
 
 					incorrectProcesses := fdbv1beta2.FilterByCondition(cluster.Status.ProcessGroups, fdbv1beta2.IncorrectCommandLine, false)
-					Expect(incorrectProcesses).To(Equal([]fdbv1beta2.ProcessGroupID{storageOneProcessGroupID}))
-
-					Expect(len(cluster.Status.ProcessGroups)).To(BeNumerically(">", 4))
-					processGroup := cluster.Status.ProcessGroups[len(cluster.Status.ProcessGroups)-4]
-					Expect(processGroup.ProcessGroupID).To(Equal(storageOneProcessGroupID))
-					Expect(len(processGroup.ProcessGroupConditions)).To(Equal(1))
-					Expect(processGroup.IsMarkedForRemoval()).To(BeTrue())
```

Comment:
the check on 412 doesn't seem superfluous to me (not in the filterByCondition like previous ones)